taskstats/src/message.rs

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+
+//! Generic netlink controller implementation
+//!
+//! This module provides the definition of the taskstats packet.
+
+use crate::nlas::*;
+use anyhow::Context;
+use netlink_packet_generic::{GenlFamily, GenlHeader};
+use netlink_packet_utils::{nla::NlasIterator, traits::*, DecodeError};
+use std::convert::{TryFrom, TryInto};
+use serde::{Serialize, Deserialize};
+
+/// Netlink attributes for this family
+
+/// Command code definition 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum TaskStatsCmdCodes {
+    /// user->kernel request/get-response
+	Get,
+    /// kernel->user event
+	New,
+}
+
+pub const TASKSTATS_CMD_GET: u8 = 1;
+pub const TASKSTATS_CMD_NEW: u8 = 2;
+
+impl From<TaskStatsCmdCodes> for u8 {
+    fn from(cmd: TaskStatsCmdCodes) -> u8 {
+        use TaskStatsCmdCodes::*;
+        match cmd {
+            Get => TASKSTATS_CMD_GET,
+            New => TASKSTATS_CMD_NEW,
+        }
+    }
+}
+
+impl TryFrom<u8> for TaskStatsCmdCodes {
+    type Error = DecodeError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        use TaskStatsCmdCodes::*;
+        Ok(match value {
+            TASKSTATS_CMD_GET => Get,
+            TASKSTATS_CMD_NEW => New,
+            cmd => {
+                return Err(DecodeError::from(format!(
+                    "Unknown taskstat command: {}",
+                    cmd
+                )))
+            }
+        })
+    }
+}
+
+
+/// Payload of taskstats
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct TaskStatsCmd {
+    /// Command code of this message
+    pub cmd: TaskStatsCmdCodes,
+    /// Netlink attributes in this message
+    pub nlas: Vec<TaskStatsCmdAttrs>,
+    /// family id is not fixed
+    pub family_id: u16
+}
+
+impl GenlFamily for TaskStatsCmd {
+    fn family_name() -> &'static str {
+        "taskstats"
```

Comment:
The kernel has:

```c
#define TASKSTATS_GENL_NAME     "TASKSTATS"
```

Can you confirm your code is actual working?

Reply:
Fixed it anyhow.